pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -1006,9 +1259,131 @@ func (r *ReconcileClusterPool) createCloudBuilder(pool *hivev1.ClusterPool, logg
+	var inventory ClusterDeploymentCustomizations
+	for _, entry := range pool.Spec.Inventory {
+		if entry.Kind == hivev1.ClusterDeploymentCustomizationInventoryEntry || entry.Kind == "" {
+			cdc := &hivev1.ClusterDeploymentCustomization{}
+			if err := r.Client.Get(context.Background(), client.ObjectKey{Namespace: pool.Namespace, Name: entry.Name}, cdc); err != nil {
+				if apierrors.IsNotFound(err) {
+					r.setInventoryValidCondition(pool, true, cdc.Name, "missing", logger)
+				}
+				continue
+			}
+			currentAvailability := controllerutils.FindClusterDeploymentCustomizationCondition(
+				cdc.Status.Conditions,
+				hivev1.ClusterDeploymentCustomizationAvailableCondition,
+			)
+			if currentAvailability == nil || currentAvailability.Status == corev1.ConditionTrue {
+				inventory = append(inventory, *cdc)
+			}
+		}
+	}
+	if inventory.Len() > 0 {
+		sort.Sort(inventory)
+		cdc := &inventory[0]
+		conds, changed := controllerutils.SetClusterDeploymentCustomizationCondition(
+			cdc.Status.Conditions,
+			hivev1.ClusterDeploymentCustomizationAvailableCondition,
+			corev1.ConditionFalse,
+			"Reservation",
+			"Reserving cluster deployment customization",
+			controllerutils.UpdateConditionIfReasonOrMessageChange,
+		)
+		if changed {
+			cdc.Status.Conditions = conds
+			if err := r.Status().Update(context.Background(), cdc); err != nil {
+				if apierrors.IsNotFound(err) {
+					r.setInventoryValidCondition(pool, true, cdc.Name, "missing", logger)
+				}
+				return nil, errors.New("could not update ClusterDeploymentCustomization conditions")
+			}
+		}
+
+		return cdc, nil
+	}
+
+	return nil, errors.New("no customization available")
+}
+
+func applyPatches(patches []hivev1.PatchEntity, data string, logger log.FieldLogger) (string, error) {
+	targetJson, err := yaml.YAMLToJSON([]byte(data))
+	if err != nil {
+		log.WithError(err).Error("unable to parse install-config template")
+		return data, err
+	}
+
+	patchJson, err := json.Marshal(patches)
+	if err != nil {
+		log.WithError(err).Error("unable to marshal patches to json")
+		return data, err
+	}
+
+	patch, err := jsonpatch.DecodePatch(patchJson)
+	if err != nil {
+		log.WithError(err).Error("unable to create json patch")
+		return data, err
+	}
+
+	patchedJson, err := patch.Apply(targetJson)
+	if err != nil {
+		log.WithError(err).Error("unable to patch install-config template")
+		return data, err
+	}
+
+	patchedYaml, _ := yaml.JSONToYAML(patchedJson)
+
+	return string(patchedYaml), nil
+}
+
+func isInstallConfigSecret(obj interface{}) bool {
+	if secret, ok := obj.(*corev1.Secret); ok {
+		_, ok := secret.StringData["install-config.yaml"]
+		if ok {
+			return true
+		}
+	}
+	return false
+}
+
+// ClusterDeploymentCustomizations is a list ClusterDeploymentCustomization objects
+type ClusterDeploymentCustomizations []hivev1.ClusterDeploymentCustomization
```

Comment:
(For your awareness, in case you felt like it was silly to create this typedef and vacuous `Len()`/`Swap()` methods just to satisfy the sort Interface, you could use [Slice()](https://pkg.go.dev/sort#Slice) instead. Basically you pass the slice and your `Less` func as parameters and you're done. [Example](https://github.com/openshift/hive/blob/ea63cac7a85032083993849513e53ffe65b3fe83/pkg/controller/clusterpool/clusterpool_controller.go#L542-L552).)